src/sqlancer/mysql/oracle/MySQLCERTOracle.java

```diff
@@ -0,0 +1,254 @@
+package sqlancer.mysql.oracle;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import sqlancer.IgnoreMeException;
+import sqlancer.Randomly;
+import sqlancer.SQLGlobalState;
+import sqlancer.common.DBMSCommon;
+import sqlancer.common.oracle.TestOracle;
+import sqlancer.common.query.ExpectedErrors;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.common.query.SQLancerResultSet;
+import sqlancer.mysql.MySQLErrors;
+import sqlancer.mysql.MySQLGlobalState;
+import sqlancer.mysql.MySQLSchema.MySQLTables;
+import sqlancer.mysql.MySQLVisitor;
+import sqlancer.mysql.ast.MySQLBinaryLogicalOperation;
+import sqlancer.mysql.ast.MySQLBinaryLogicalOperation.MySQLBinaryLogicalOperator;
+import sqlancer.mysql.ast.MySQLColumnReference;
+import sqlancer.mysql.ast.MySQLExpression;
+import sqlancer.mysql.ast.MySQLSelect;
+import sqlancer.mysql.ast.MySQLTableReference;
+import sqlancer.mysql.gen.MySQLExpressionGenerator;
+
+public class MySQLCERTOracle implements TestOracle<MySQLGlobalState> {
+    private final MySQLGlobalState state;
+    private MySQLExpressionGenerator gen;
+    private final ExpectedErrors errors = new ExpectedErrors();
+    private MySQLSelect select;
+    private List<String> queryPlan1Sequences;
+    private List<String> queryPlan2Sequences;
+
+    public enum Mutator {
```

Comment:
Could the enum be shared among different implementations? If so, perhaps, this could be called `CertMutation`.